src/OpenSage.Game/Client/Drawable.cs

```diff
@@ -767,3 +2098,77 @@ public enum ObjectDecalType
     None = 8,
     ShadowTexture = 9,
 }
+
+public enum StealthLookType
+{
+    /// <summary>
+    /// Unit is not stealthed at all.
+    /// </summary>
+    None = 0,
+
+    /// <summary>
+    /// Unit is stealthed-but-visible due to friendly status.
+    /// </summary>
+    VisibleFriendly = 1,
+
+    /// <summary>
+    /// We can have units that are disguised (instead of invisible).
+    /// </summary>
+    DisguisedEnemy = 2,
+
+    /// <summary>
+    /// Unit is stealthed and invisible, but a second material pass
+    /// is added to reveal the invisible unit as with heat vision.
+    /// </summary>
+    VisibleDetected = 3,
+
+    /// <summary>
+    /// Unit is stealthed-but-visible due to being detected,
+    /// and rendered in heatvision effect second material pass.
+    /// </summary>
+    VisibleFriendlyDetected = 4,
+
+    /// <summary>
+    /// Unit is stealthed-and-invisible.
+    /// </summary>
+    Invisible = 5,
+}
+
+/// <summary>
+/// Simple structure used to bind W3D render objects to our own Drawables.
+/// </summary>
+public class DrawableInfo
+{
+    [Flags]
+    public enum ExtraRenderFlags
+    {
+        None = 0,
+        IsOccluded = 0x1,
+        PotentialOccluder = 0x2,
+        PotentialOccludee = 0x4,
+        IsTranslucent = 0x8,
+        IsNonOccluderOrOccludee = 0x10,
+        DelayedRender = IsTranslucent | PotentialOccludee,
```

Comment:
I'm guessing the way this will end up being used is like `if (ExtraRenderFlags.DelayedRender.HasFlag(variable))` since I don't think it works the other way around, which I could see being confusing.

I guess what I'm saying is we might have to be careful about how this is used. If nothing is ever _set_ to `DelayedRender` and it would only be used to check if something is of this type (and assuming if it has _either_ value it should be delayed), maybe an extension method would be better?

Or maybe that's not how it will be used at all 🤷

Reply:
I agree, it would be nicer with an extension method. cf776357ec7d2ca4572d63f9be9dcdeab330d946